src/installer/sources.py

```diff
@@ -287,11 +287,11 @@ def validate_record(self, *, validate_contents: bool = True) -> None:
                     f"In {self._zipfile.filename}, hash / size of {item.filename} is not included in RECORD"
                 )
             if validate_contents:
-                data = self._zipfile.read(item)
-                if not record.validate(data):
-                    issues.append(
-                        f"In {self._zipfile.filename}, hash / size of {item.filename} didn't match RECORD"
-                    )
+                with self._zipfile.open(item, "r") as stream:
+                    if not record.validate(cast("BinaryIO", stream), item.file_size):
```

Comment:
(straying into changes beyond the scope of this MR: but so far as I can see this repository could use `IO[bytes]` everywhere it currently uses `BinaryIO` which would save some casting)